pyanaconda/core/configuration/system.py

```diff
@@ -125,7 +131,7 @@ def can_activate_layouts(self):
     @property
     def can_configure_keyboard(self):
         """Can we configure the keyboard?"""
-        return self._is_boot_iso or self._is_live_os or self._is_booted_os
+        return (not self._is_wayland) and (self._is_boot_iso or self._is_live_os or self._is_booted_os)
```

Comment:
I guess the proper solution would be to expose `can_configure_keyboard` in the configuration files and set it up in the `AnacondaConfiguration.set_from_opts` method if that's possible. The config would stay simple and stable. However, the current solution seems to be good enough as a temporary workaround.

Reply:
Not sure I understand you, I don't plan to use opts here. I want to detect Wayland and that is not doable from the opts.

Reply:
I wonder wouldn't it be better to instead of changing the can_configure_keyboard in set_from_opts add is_wayland?